examples/with-okta-multi-tenant-pkce-flow/api-server/okta.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/supertokens/supertokens-golang/recipe/thirdparty/tpmodels"
+	"github.com/supertokens/supertokens-golang/supertokens"
+)
+
+type OktaConfig struct {
+	ClientID   string
+	OktaDomain string
+}
+
+var oktaProvider = tpmodels.TypeProvider{
+	ID:        "okta",
+	IsDefault: true,
+	Get: func(redirectURI, authCodeFromRequest *string, userContext supertokens.UserContext) tpmodels.TypeProviderGetResponse {
+		tenantId := getTenantIdFromUserContext(userContext)
+		config := getConfigForTenantId(tenantId)
+
+		oktaRedirectURI := "http://multitenant.com:3000/auth/callback/okta" // TODO: Update as per your website domain
+
+		// Optional: Adding tenantId to the user context so that it can be added to accessTokenPayload while creating new session
+		(*userContext)["tenantId"] = tenantId
+
+		if authCodeFromRequest == nil {
+			empty := ""
+			authCodeFromRequest = &empty
+		}
+
+		state := getStateFromUserContext(userContext)
```

Comment:
Explain why this state is needed and why it's coming from the frontend.

Reply:
done